packages/sources/finnhub-test/src/endpoint/quote.ts

```diff
@@ -0,0 +1,51 @@
+import { EndpointTypes } from './router'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+
+export const commonKeys: Record<string, string> = {
+  N225: '^N225',
+  FTSE: '^FTSE',
+  XAU: 'OANDA:XAU_USD',
+  XAG: 'OANDA:XAG_USD',
+  AUD: 'OANDA:AUD_USD',
+  EUR: 'OANDA:EUR_USD',
+  GBP: 'OANDA:GBP_USD',
+  // CHF & JPY are not supported
+}
+
+export const httpTransport = new HttpTransport<EndpointTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      let symbol = param.base.toUpperCase()
+      if (commonKeys[symbol]) {
+        symbol = commonKeys[symbol]
+      }
+
+      const requestConfig = {
+        baseURL: `${config.API_ENDPOINT}/quote`,
+        method: 'GET',
+        params: {
+          symbol,
+          token: config.API_KEY,
+        },
+      }
+      return {
+        params,
+        request: requestConfig,
+      }
+    })
+  },
+  parseResponse: (params, res) => {
+    return params.map((param) => {
+      const result = res.data.c
```

Comment:
Could you add a validation that checks if c has a value here? And return an error if it doesn't

Reply:
Done